xarray/core/ops.py

```diff
@@ -260,6 +260,9 @@ def inject_reduce_methods(cls):
         + [("count", duck_array_ops.count, False)]
     )
     for name, f, include_skipna in methods:
+        if hasattr(cls, name):
+            name = "_injected_" + name
```

Comment:
We're only using `_injected_*` for two methods in this file (argmin/argmax), but you added it this check everywhere methods get added. This makes this already complicated modules more complicated :/.

I would strongly prefer to avoid adding these `_injected_argmin` methods if possible. Let me see if I can work-up an example for you...

Reply:
OK, I just pushed a commit to your branch that shows how you can call `Variable.reduce()` instead of using these injected methods. I think it should be possible to remove all of these injected methods now.